samples/SocialSample/Startup.cs

```diff
@@ -199,13 +206,10 @@ 127.0.0.1 MsSecSample.localhost.this
                         await context.Response.WriteAsync("<a href=\"?authscheme=" + type.AuthenticationScheme + "\">" + (type.Caption ?? "(suppressed)") + "</a><br>");
                     }
                     await context.Response.WriteAsync("</body></html>");
-                });
-            });
-
-            // Sign-out to remove the user cookie.
-            app.Map("/logout", signoutApp =>
-            {
-                signoutApp.Run(async context =>
+                    return;
+                }
+                
+                if (context.Request.Path.StartsWithSegments(new PathString("/logout")))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
                     context.Response.ContentType = "text/html";
```

Comment:
I'm not saying that the new behavior is necessarily bad, it's just **surprising** and... obviously different :smile:

What I'm saying is that we can't ship these new methods without changing the way we use them: altering the headers after invoking `SignInAsync/SignOutAsync/ChallengeAsync` is now very risky, specially if the authentication handler started writing to the response stream.

For instance, this snippet will throw an exception:

```csharp
await Context.Authentication.SignInAsync(OpenIdConnectDefaults.AuthenticationScheme, new ClaimsPrincipal(identity));

return new HttpStatusCodeResult(200);
```
... while this one, that uses `EmptyResult` (which doesn't change the status code), won't:

```csharp
await Context.Authentication.SignInAsync(OpenIdConnectDefaults.AuthenticationScheme, new ClaimsPrincipal(identity));

return new EmptyResult();
```

![image](https://cloud.githubusercontent.com/assets/6998306/8386400/471351a6-1c50-11e5-837c-72c60d090121.png)